daft/runners/progress_bar.py

```diff
@@ -16,12 +16,32 @@ def __init__(self, use_ray_tqdm: bool, show_tasks_bar: bool = False, disable: bo
         self.use_ray_tqdm = use_ray_tqdm
         if use_ray_tqdm:
             from ray.experimental.tqdm_ray import tqdm
-
-            self.tqdm_mod = tqdm
         else:
-            from tqdm.auto import tqdm
+            from tqdm.auto import tqdm as _tqdm
+
+            try:
+                import sys
+
+                from IPython import get_ipython
+
+                ipython = get_ipython()
+
+                # write to sys.stdout if in jupyter notebook
+                if ipython is not None and "IPKernelApp" in ipython.config:
```

Comment:
Let's document where we got this logic from? It's not obvious to me at first glance that this is a canonical/foolproof way of checking if we are running in a jupyter notebook